scripts/generate-lint.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	lintPrefix        = "lint"
+	lintVariant       = "race-detector"
+	lintGroup         = "lint-group"
+	groupMaxHosts     = 1
+	evergreenLintTask = "evergreen"
+	jsonFilename      = "generate-lint.json"
```

Comment:
arguably these should be parsed using flags.

Reply:
I don't think so. Passing other values for many of these is not meaningful, since the script doesn't generalize particularly well. A few are meaningful, but changing constants is okay, since it's a script. If we end up using more generate in self-tests.yml, we can make the script more general, and use flags.